pylon/src/nginx.conf.template

```diff
@@ -74,26 +97,99 @@ http {
     }
 
     #
-    # Web portals
+    # YARN web portal
+    #
+
+    location ~ ^/yarn$ {
+      # Add '/' to the end of the URL, otherwise there will be a 404 error.
+      return 301 $scheme://$http_host$request_uri/;
+    }
+    location ~ ^/yarn/([^/]+)/(\d+)/(.*/)?([^/]*)$ {
```

Comment:
I'm thinking of we should limit the user's freedom to compose Yarn urls with arbitrary target_port so that users will not able to use this proxy to probe our backend resources randomly but only able to visit RMs and NMs.

And also we should like WebHDFS not let user to specify RM host name but we pick it up from environment for not exposing deployment specific stuff on API path.

Reply:
Definitely. But this requires a redesign of the service configuration mechanism, and also some fundamental addition to the nginx configuration file. I'll create a ticket for this feature.